src/Objects/Monitor.vala

```diff
@@ -24,6 +24,12 @@ public class Display.Monitor : GLib.Object {
     public string vendor { get; set; }
     public string product { get; set; }
     public string serial { get; set; }
+    public uint hash {
+        get {
+            return (connector+vendor+product+serial).hash ();
+        }
+    }
+    
```

Comment:
Should not the Monitor object just have an immutable MonitorInfo struct? The vendor, product etc should not really be mutable variables - they correspond to real hardware.